### PR TITLE
Flinguenheld

### PR DESCRIPTION
**Pull request by FLinguenheld**
## Description

This is my custom keymap for the aurora sweep keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/README.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
+### FLinguenheld custom aurora sweep keymap
```
Comment (waffle87) [inline keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/README.md:1]: This filename should be lowercase.

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/rules.mk [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Recommended by the doc to improve compilation :
+LTO_ENABLE = yes
+SPLIT_KEYBOARD = yes
+
+
+BOOTLOADER = rp2040
```
Comment (waffle87) [inline keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/rules.mk:6]:
```suggestion
```

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/features/auto_shift.c [hunk 3]
```diff
@@ -0,0 +1,193 @@
+/* Only for basis letters
```
Comment (waffle87) [inline keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/features/auto_shift.c:1]: This file is missing a license header.
eg.
```
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/features/combo.c [hunk 4]
```diff
@@ -0,0 +1,264 @@
+/* How to :
```
Comment (waffle87) [inline keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/features/combo.c:1]: This file is missing a license header.

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/features/leader.c [hunk 5]
```diff
@@ -0,0 +1,263 @@
+LEADER_EXTERNS();
```
Comment (waffle87) [inline keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/features/leader.c:1]: This file is missing a license header.

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/features/oled.c [hunk 6]
```diff
@@ -0,0 +1,118 @@
+#include "../font/logos.c"
```
Comment (waffle87) [inline keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/features/oled.c:1]: C source files should not be `include`'d. Include them via `SRC +=` in your rules.mk file.

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/features/oled.c [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
```
Comment (waffle87) [inline keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/features/oled.c:2]: This file is missing a license header.

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/font/glcdfont.c [hunk 8]
```diff
@@ -0,0 +1,229 @@
+#include "progmem.h"
```
Comment (waffle87) [inline keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/font/glcdfont.c:1]: This file is missing a license header.

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/keymap.c [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+    CS_Y_DIAERESIS,
+};
+
+/* Features */
+#include "features/auto_shift.c"
+#include "features/combo.c"
+#include "features/leader.c"
+#include "features/oled.c"
+
+
```
Comment (waffle87) [inline keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/keymap.c:66]: Above comment regarding includes.
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/features/oled.c (hunk 6 above)
Comment (FLinguenheld) [reply keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/features/oled.c]: I'm getting crazy with includes ^^`
Is it correct to put them in the keymap.c ?

**Review by drashna — APPROVED** [review]

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/keymap.c [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+#include QMK_KEYBOARD_H
+#include <stdio.h>
+
+#include "keycodes.h"
+#include "features/auto_shift.c"
+#include "features/combo.c"
+#include "features/leader.c"
+#include "oled/logos.c"
+#include "oled/oled.c"
```
Comment (drashna) [inline keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/keymap.c:25]: These should be added via rules.mk, and `SRC += (file names)`.

If needed, header files should be used included (eg, `auto_shift.h`)

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/keymap.c [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include <stdio.h>
```
Comment (drashna) [inline keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/keymap.c:18]:
```suggestion
```

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/keycodes.h [hunk 12]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 Florent Linguenheld (@FLinguenheld)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
```
Comment (waffle87) [inline keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/keycodes.h:3]:
```suggestion
#pragma once

```

**Review by waffle87 — APPROVED** [review]

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/rules.mk [hunk 13]
```diff
@@ -0,0 +1,23 @@
+LTO_ENABLE = yes
+SPLIT_KEYBOARD = yes
```
Comment (drashna) [inline keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/rules.mk:2]: This isn't needed
```suggestion
```

**Review by drashna — APPROVED** [review]

File: keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/rules.mk (hunk 13 above)
Comment (FLinguenheld) [reply keyboards/splitkb/aurora/sweep/keymaps/flinguenheld/rules.mk]: Thank you 👌

**Pull request merged**